x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/original_scenarios.md

```diff
@@ -0,0 +1,264 @@
+Then no callout message is displayed for <prebuilt_rules_status> rules
+And no badges with number of available rules to install/update are displayed
+
+Examples:
+  | prebuilt_rules_status |
+  |            to install |
+  |             to update |
+```
+
+#### **Scenario: Callout message is displayed when there are new prebuilt rules available to install**
+
+```Gherkin
+Given user already has 8.x prebuilt rules installed
+And there are new prebuilt rules available to install
+When user navigates to the Rules Management Page
+Then user should see a callout message to install new prebuilt rules
+And the number of new rules available to install should be displayed on the +Add Elastic Rules link
+```
+
+#### **Scenario: Callout message is displayed when there are new updates on already installed prebuilt rules**
+
+```Gherkin
+Given user already has 8.x prebuilt rules installed
+And there are new updates available for those prebuilt rules
+When users navigate to the Rules Management Page
+Then users should see an update callout message
+And the number of outdated rules should be displayed on the Rules Updates tab
+```
+
+#### **Scenario: User is notified of available prebuilt rules to install when a rule is deleted**
+
+```Gherkin
+Given user has the latest version of prebuilt rules installed
+When user navigates to Rules Management Page
+And user deletes some prebuilt rules
+Then user should see a callout message with the same amount of prebuilt rules ready to install
+And the updated number of available rules to install should be displayed on the +Add Elastic Rules link
+```
+
+#### **Scenario: User is notified that all available rules have been installed in the Add Elastic Rules**
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Add Elastic Rules Page
+Then user should see a message indicating that all available rules have been installed
+And user should see a CTA that leads to the Rules Management Page
+And user should navigate back to Rules Management Page when clicking on the CTA
+```
+
+#### **Scenario: User is notified that all installed rules are up to date in the Rule Updates tab**
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Rule Update
+Then user should see a message indicating that all installed rules are up to date
+```
+
+### Prebuilt Rules Installation
+
+#### **Scenario: User without any installed prebuilt rules can install `<amount>` prebuilt rules**
+
+```Gherkin
+Given a user that doesn’t have prebuilt rules installed
+When user navigates to Add Elastic Rules Page
+Then available prebuilt rules are displayed on Elastic Rules table
+And user can install <amount> prebuilt Rules
+And successfully installed message is displayed after installation
+And installed rules are removed from Elastic Rules table
+And rules to install counter is decreased accordingly
+
+Examples:
+  | amount     |
+  | all        |
+  | selected   |
+  | individual |
+```
```

Comment:
Split this into 3 scenarios to be able to add more details into them:

#### **Scenario: User can install prebuilt rules one by one**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given no prebuilt rules are installed in Kibana
And there are X prebuilt rules available to install
When user opens the Add Rules page
Then prebuilt rules available for installation should be displayed in the table
When user installs one individual rule
Then success message should be displayed after installation
And the installed rule should be removed from the table
When user navigates back to the Rule Management page
Then user should see a CTA to install prebuilt rules
And user should see the number of rules available to install decreased by 1
```

#### **Scenario: User can install multiple prebuilt rules selected on the page**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given no prebuilt rules are installed in Kibana
And there are X prebuilt rules available to install
When user opens the Add Rules page
Then prebuilt rules available for installation should be displayed in the table
When user selects <Y> rules
Then user should see a CTA to install <Y> number of rules
When user clicks the CTA
Then success message should be displayed after installation
And all the installed rules should be removed from the table
When user navigates back to the Rule Management page
Then user should see a CTA to install prebuilt rules
And user should see the number of rules available to install decreased by <Y> number of installed rules

Examples:
  | Y                               |
  | a few rules on the page, e.g. 2 |
  | all rules on the page, e.g. 12  |
```

#### **Scenario: User can install all available prebuilt rules at once**

**Coverage**: 1 e2e test with mock rules.

```Gherkin
Given no prebuilt rules are installed in Kibana
And there are X prebuilt rules available to install
When user opens the Add Rules page
Then prebuilt rules available for installation should be displayed in the table
When user installs all rules
Then success message should be displayed after installation
And all the rules should be removed from the table
And user should see a message indicating that all available rules have been installed
And user should see a CTA that leads to the Rule Management page
When user clicks on the CTA
Then user should be navigated back to Rule Management page
And user should NOT see a CTA to install prebuilt rules
And user should NOT see a number of rules available to install
```